app/src/main/java/com/likeminds/feedsx/post/detail/view/PostDetailFragment.kt

```diff
@@ -261,6 +249,74 @@ class PostDetailFragment :
         postDetailExtras = arguments?.getParcelable(POST_DETAIL_EXTRAS)!!
     }
 
+    // processes delete entity request
+    private fun deleteEntity(
+        entityId: String,
+        @ReportType
+        entityType: Int
+    ) {
+        //TODO: set isAdmin
+        val isAdmin = true
+        if (isAdmin) {
+            val deleteEntityExtras = DeleteEntityExtras.Builder()
+                .entityId(entityId)
+                .entityType(entityType)
+                .build()
+            DeleteEntityDialogFragment.showDialog(
+                childFragmentManager,
+                deleteEntityExtras
+            )
+        } else {
+            showDeleteEntityDialog(entityType)
+        }
+    }
+
+    // shows delete entity dialog when user deletes their own entity (post/comment)
+    private fun showDeleteEntityDialog(
+        @ReportType
+        entityType: Int
+    ) {
+        val builder = AlertDialog.Builder(requireContext())
+        var message = getString(R.string.delete_post_message)
+        var title = getString(R.string.delete_post_question)
+        if (entityType == REPORT_TYPE_COMMENT) {
+            message = getString(R.string.delete_comment_message)
+            title = getString(R.string.delete_comment_question)
+        }
+        builder.setMessage(message)
+            .setTitle(title)
+            .setCancelable(true)
+            .setPositiveButton(getString(R.string.delete_caps)) { _, _ ->
+
+            }.setNegativeButton(getString(R.string.cancel_caps)) { _, _ ->
+                alertDialog.dismiss()
+            }
+        //Creating dialog box
+        alertDialog = builder.create()
+        alertDialog.show()
+        alertDialog.getButton(AlertDialog.BUTTON_NEGATIVE)
+            ?.setTextColor(ContextCompat.getColor(requireContext(), R.color.black_40))
```

Comment:
dont create dialog using this method. create a new class using BaseDialogFragment and show it here